app/view_objects/allocations_view.rb

```diff
@@ -0,0 +1,64 @@
+class AllocationsView
+  module Status
+    REQUESTED = "REQUESTED".freeze
+    NOT_REQUESTED = "NOT REQUESTED".freeze
+    YET_TO_REQUEST = "YET TO REQUEST".freeze
+  end
+
+  module Colour
+    GREEN = "green".freeze
+    RED = "red".freeze
+    GREY = "grey".freeze
+  end
+
+  module Requested
+    YES = "yes".freeze
+    NO = "no".freeze
+  end
+
+  def initialize(training_providers:, allocations:)
+    @training_providers = training_providers
+    @allocations = allocations
+  end
+
+  def allocation_statuses
+    @training_providers.map do |training_provider|
+      matching_allocation = find_matching_allocation(training_provider)
+      build_allocation_status(matching_allocation, training_provider)
+    end
+  end
+
+private
+
+  def find_matching_allocation(training_provider)
+    @allocations.find do |allocation|
+      allocation.provider.id == training_provider.id
+    end
+  end
+
+  def build_allocation_status(matching_allocation, training_provider)
+    allocation_status = {
+      training_provider_name: training_provider.provider_name,
+      training_provider_code: training_provider.provider_code,
+    }
+
+    if matching_allocation.nil?
```

Comment:
We could move each of these predicates into a method e.g. `yet_to_request?`, `requested?` and `not_requested?` then it would be a bit clearer what their significance is and it would isolate the changes we're going to need to make when the `request_type` gets added to the response. We could do that when we make the changes though so not a biggy.